snaps_tests/demos_tests/test_gradle_with_outputdir.py

```diff
@@ -0,0 +1,29 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import snaps_tests
+
+
+class GradleTestCase(snaps_tests.SnapsTestCase):
+
+    snap_content_dir = 'gradle-with-outputdir'
+
+    def test_java_hello_world(self):
```

Comment:
this would need a similar construct to the one in https://github.com/snapcore/snapcraft/blob/master/snaps_tests/demos_tests/test_gradle.py#L27

Or fix the armhf issue we have which would also be a welcome addition!